pkg/kubelet/kubelet.go

```diff
@@ -1112,7 +1118,7 @@ func (kl *Kubelet) pullImageAndRunContainer(pod *api.Pod, container *api.Contain
 type podContainerChangesSpec struct {
 	startInfraContainer bool
 	infraContainerId    dockertools.DockerID
-	containersToStart   map[int]empty
+	containersToStart   map[int]string
```

Comment:
I am not too happy about hiding the reason to start in `podContainerChangesSpec`, since it's named "ChangesSpec". But we need to convey the reason in addition to the spec, and this is the least invasive way. A cleaner way may be to add a separate return object.